### ISSUE TITLE
[BUG] AKS with CBL-Mariner image can't pull image from ACR due to error "x509: certificate signed by unknown authority"

### ISSUE THREAD

**Issue by vlardn** [Azure/AKS#3320]
**Describe the bug**
AKS created with "--os-sku CBLMariner" option can't pull images from attached ACR

**To Reproduce**
1. Run commands
```
RG_NAME=mariner-tst
RG_LOCATION=EastUS
ACR_NAME=marinertst
AKS_NAME=mariner-tst
az group create --name $RG_NAME --location $RG_LOCATION > /dev/null
az acr create --name $ACR_NAME --resource-group $RG_NAME --sku Basic > /dev/null
az aks create --name $AKS_NAME --resource-group $RG_NAME --os-sku CBLMariner --attach-acr $ACR_NAME --generate-ssh-keys > /dev/null
az aks check-acr --name $AKS_NAME --resource-group $RG_NAME --acr "$ACR_NAME.azurecr.io"
```
2. See error
```
[2022-11-10T06:58:09Z] Checking host name resolution (marinertst.azurecr.io): SUCCEEDED
[2022-11-10T06:58:09Z] Canonical name for ACR (marinertst.azurecr.io): r1007eus-3.eastus.cloudapp.azure.com.
[2022-11-10T06:58:09Z] ACR location: eastus
[2022-11-10T06:58:09Z] Checking managed identity...
[2022-11-10T06:58:09Z] Kubelet managed identity client ID: 98a47625-fe05-423c-9a10-a368ae9208bc
[2022-11-10T06:58:09Z] Validating managed identity existance: SUCCEEDED
[2022-11-10T06:58:09Z] Validating image pull permission: FAILED
[2022-11-10T06:58:09Z] ACR marinertst.azurecr.io rejected token exchange: failed to send token exchange request: Post "https://marinertst.azurecr.io/oauth2/exchange": x509: certificate signed by unknown authority
```

**Expected behavior**
Removing "--os-sku CBLMariner" option from "az aks create" command above (i.e., using Ubuntu) results in no error:
```
[2022-11-10T07:15:43Z] Checking host name resolution (marinertst.azurecr.io): SUCCEEDED
[2022-11-10T07:15:43Z] Canonical name for ACR (marinertst.azurecr.io): r1007eus-3.eastus.cloudapp.azure.com.
[2022-11-10T07:15:43Z] ACR location: eastus
[2022-11-10T07:15:43Z] Checking managed identity...
[2022-11-10T07:15:43Z] Kubelet managed identity client ID: 8591bfec-3c6f-4b39-8a7a-0d9f143b81b5
[2022-11-10T07:15:44Z] Validating managed identity existance: SUCCEEDED
[2022-11-10T07:15:44Z] Validating image pull permission: SUCCEEDED
[2022-11-10T07:15:44Z]
Your cluster can pull images from marinertst.azurecr.io!
```

**Environment:**
 - Kubernetes version - 1.23.12
 - Azure CLI version 2.42.0

**Comment by wangyira** [comment]
@alexeldeib could you take a look at this issue?

**Comment by CocoWang-wql** [comment]
Hello @vlardn can you pls run the command `az aks get-credentials -g <RESOURCE_GROUP_NAME> -n <CLUSTER_NAME> --overwrite-existing` and try the steps again?

**Comment by vlardn** [comment]
Hi @CocoWang-wql 
Just re-tested all steps. The problem still exists:
```
$ az aks check-acr --name $AKS_NAME --resource-group $RG_NAME --acr "$ACR_NAME.azurecr.io"
Merged "mariner-tst" as current context in /tmp/tmp2y0f82g4
[2023-03-09T13:06:53Z] Checking host name resolution (marinertst.azurecr.io): SUCCEEDED
[2023-03-09T13:06:53Z] Canonical name for ACR (marinertst.azurecr.io): r0224eus-3.eastus.cloudapp.azure.com.
[2023-03-09T13:06:53Z] ACR location: eastus
[2023-03-09T13:06:53Z] Checking managed identity...
[2023-03-09T13:06:53Z] Kubelet managed identity client ID: b0a9a56c-10ae-447f-a51c-1d9a7dfaf94b
[2023-03-09T13:06:53Z] Validating managed identity existance: SUCCEEDED
[2023-03-09T13:06:53Z] Validating image pull permission: FAILED
[2023-03-09T13:06:53Z] ACR marinertst.azurecr.io rejected token exchange: failed to send token exchange request: Post "https://marinertst.azurecr.io/oauth2/exchange": x509: certificate signed by unknown authority
```

**Comment by CocoWang-wql** [comment]
It's more like the networking setting issue (VPN, firewall, etc). Can you pls open a support ticket? As you are using the acr, the support eng can help track some backend logs and requests.

**Comment by vlardn** [comment]
@CocoWang-wql  
The `x509: certificate signed by unknown authority` error can't be a networking issue. Have you tried these 4 `az` commands on any non-Microsoft corporate computer?

**Comment by alexeldeib** [comment]
@hbeberman does this ring some bells for you? It vaguely does for me, need to try a repro. AKS e2e already use private ACR but I think we don't use kubelet identity to pull for most scenarios, wonder if that code path has issues on mariner.

**Comment by hbeberman** [comment]
Hi @vlardn, thank you for raising this issue!
We root caused this to an issue where the check-acr command was incompatible with Mariner and resolved it in az-cli with this PR https://github.com/Azure/azure-cli/pull/25453

Looking at the [az-cli release notes](https://github.com/MicrosoftDocs/azure-docs-cli/blob/main/docs-ref-conceptual/release-notes-azure-cli.md#aks) the fix shipped with az-cli version 2.46.0, so please try upgrading to that version and let us know how it goes.

Additionally, I suggest that you use `--os-sku Mariner` instead of `--os-sku CBLMariner` going forward. The CBLMariner OSSKU selects an older version which will eventually be deprecated.

**Comment by vlardn** [comment]
Hi, I confirm the problem is fixed now. So issue can be closed.
